File: warmup-2_last2.java
public int last2(String str) {
  if (str.length() < 3) return 0;
  int count = 0;
  String last = str.substring(str.length() - 2);
  for (int i = 0; i < str.length() - 2; i++) {
    if (str.substring(i, i + 2).equals(last)) count++;
  }
  return count;
}

File: warmup-2_last2.py
def last2(str):
  if len(str) < 4: 
    return 0
  count = 0
  target =  str[len(str) - 2: len(str)]
  for i in range(0, len(str) - 2):
    if str[i: i + 2] == target:
      count += 1
  return count
